docs/ops/sequence/CTCLoss_4.md

```diff
@@ -0,0 +1,125 @@
+## CTCLoss <a name="CTCLoss"></a>
+
+**Versioned name**: *CTCLoss-4*
+
+**Category**: Sequence processing
+
+**Short description**: *CTCLoss* computes the CTC (Connectionist Temporal Classification) Loss.
+
+**Detailed description**:
+
+*CTCLoss* operation is presented in [Connectionist Temporal Classification - Labeling Unsegmented Sequence Data with Recurrent Neural Networks: Graves et al., 2016](http://www.cs.toronto.edu/~graves/icml_2006.pdf)
+
+*CTCLoss* estimates likelyhood that a target `labels[i,:]` can occur (or is real) for given input sequence of logits `logits[i,:,:]`.
+Briefly, *CTCLoss* operation finds all sequences aligned with a target `labels[i,:]`, computes log-probabilities of the aligned sequences using `logits[i,:,:]`
+and computes a negative sum of these log-probabilies.
+
+Input sequences of logits `logits` can have different lengths. The length of each sequence `logits[i,:,:]` equals `logit_length[i]`.
+A length of target sequence `labels[i,:]` equals `label_length[i]`. The length of the target sequence must not be greater than the length of corresponding input sequence `logits[i,:,:]`.
+Otherwise, the operation behaviour is undefined.
+
+*CTCLoss* calculation scheme:
+
+1. Compute probability of `j`-th character at time step `t` for `i`-th input sequence from `logits` using softmax formula:
+\f[
+p_{t,i,j} = \frac{\exp(logits[i,t,j])}{\sum^{K}_{k=0}{\exp(logits[i,t,k])}}
+\f]
+
+2. For a given `i`-th target from `labels[i,:]` find all aligned paths.
+A path `S = (c1,c2,...,cT)` is aligned with a target `G=(g1,g2,...,gT)` if both chains are equal after decoding.
+The decoding extracts substring of length `label_length[i]` from a target `G`, merges repeated characters in `G` in case *preprocess_collapse_repeated* equal to True and
+finds unique elements in the order of character occurence in case *unique* equal to True.
+The decoding merges repeated characters in `S` in case *ctc_merge_repeated* equal to True and removes blank characters represented by `blank_index`.
+By default, `blank_index` is equal to `C-1`, where `C` is a number of classes including the blank.
+For example, in case default *ctc_merge_repeated*, *preprocess_collapse_repeated*, *unique*, and `blank_index` a target sequence `G=(0,3,2,2,2,2,2,4,3)` of a length `label_length[i]=4` is processed 
```

Comment:
Extra comma here *unique*, and `blank_index`

Reply:
Fixed.

Reply:
Fixed.